com/paypal/Perfection.java

```diff
@@ -0,0 +1,42 @@
+package com.paypal;
+
+public class Perfection {
+
+    private static Perfection perf;
+
+    public synchronized static Perfection getPerf() {
+        if (perf == null) {
```

Comment:
I think this is will be low performance, it is going to wait until a Perfection object is being not used.